src/TapeChart.js

```diff
@@ -5,47 +5,55 @@ class TapeChart {
     this.bookings = bookings;
   }
 
-getAvailableRoomsByDate(date) {
+  getAvailableRooms(date, roomType) {
```

Comment:
This is a good cleaner way to handle this, I like how you split some of it out as a separate function.